state/state_sqlite.go

```diff
@@ -104,6 +104,51 @@ func (p *sqliteSyncState) All(epoch int64) ([]State, error) {
 	return result, nil
 }
 
+func (p *sqliteSyncState) AllByPeerID(peerID PeerID) ([]State, error) {
+	var result []State
+
+	rows, err := p.db.Query(`
+		SELECT 
+			type, send_count, send_epoch, group_id, peer_id, message_id 
```

Comment:
do we know how large this gets roughly? just maybe to make sure we don't iterate through a large amount of data